<?xml version="1.0"?>
<launch>

  <!-- Creates a Jackal in Gazebo. -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="use_sim_time" value="true"/>
    <arg name="gui"          value="false"/>
    <arg name="world_name"   value="$(find multi_jackal_tutorials)/worlds/empty.world"/>
  </include>
  <include file="$(find multi_jackal_base)/launch/jackal_base.launch">
    <arg name="ns"             value="jackal0"/>
    <arg name="use_global_tf"  value="false"/>
    <arg name="use_global_ekf" value="true"/>
    <arg name="gps_rate"       value="10"/>
    <arg name="x"              value="-90.5"/>
    <arg name="y"              value="3.5"/>
  </include>
  
  <!-- Launch formation routing. Waypoint order is [x, y, zone]. The robot will
       sit at the last point in the zone until all robots are ready, then they 
       will move to the first point in the next zone. -->
  <node ns="jackal0" pkg="formation_routing" type="formation_routing_node.py" 
        name="formation_routing" output="screen">
    <rosparam param="waypoints">[-96.5, 3.5, 0,
                                -96.8, 6.5, 0,
                                -100.0, 24.7, 0,
                                -96.5, 29.8, 0,
                                -99.1, 35.8, 0,
                                -74.4, 34.8, 0,
                                -76.2, 56.9, 0,
                                -62.1, 60.5, 0,
                                -62.7, 68.0, 0,
                                -72.3, 69.5, 0,
                                -99.1, 74.7, 0,
                                -94.1, 85.7, 0,
                                -68.1, 91.8, 0,
                                -52.0, 95.6, 0,
                                -37.0, 87.4, 0,
                                -39.1, 77.6, 0,
                                -35.8, 72.2, 0,
                                -17.6, 96.6, 0,
                                -7.4, 68.0, 0,
                                -2.7, 45.3, 0,
                                -16.0, 40.5, 0,
                                -34.4, 50.9, 0,
                                -28.8, 32.5, 0,
                                -22.2, 23.8, 0,
                                -5.1, 21.8, 0,
                                -18.7, 7.6, 0,
                                -44.3, 7.3, 0,
                                -54.2, 25.3, 0,
                                -75.4, 15.2, 0,
                                -85.5, 19.1, 0,
                                -86.2, 9.9, 0,
                                -96.5, 3.5, 0]</rosparam>
    <param name="robot_id"           value="0"/>
    <param name="baselink_frame"     value="jackal0/base_link"/>
    <param name="lin_vel"            value="2"/>
    <param name="ang_vel"            value="3"/>
    <remap from="/jackal_velocity_controller/cmd_vel" to="jackal_velocity_controller/cmd_vel"/>
  </node>
  
  <!-- Plot the paths. -->
  <node ns="jackal0" pkg="plot_path" type="plot_path_node.py"
        name="plot_path" output="screen">
    <param name="frame_from" value="map"/>
    <param name="frame_to"   value="jackal0/base_link"/>
  </node>
     
  <!-- Create a transformation. -->
  <!--<node pkg="tf" type="static_transform_publisher" name="static_map_odom_tf"
        args="0 0 0 0 0 0 map base_link 100"/>-->
  
  <!-- Display on RVIZ. -->
  <group if="true">
    <node name="rviz" pkg="rviz" type="rviz"
        args="-d $(find formation_routing)/config/rviz12.rviz">
    </node>
  </group>  
  
</launch>
